### PR TITLE
update gpg guide + more

### PR DESCRIPTION
**Pull request by kradan**
Please review these changes related to #451 

The key configuration section has been moved from openpgp-best-practices to gpg-keys (as suggested in #147 and #453). I updated all links to the old location shown by grep.

### CODE REVIEW

**Comment by kradan** [comment]
Did anybody review this PR already? From my side it's finished, let me know if anything should be corrected.

**Comment by baldurmen** [comment]
I'll try to review all of this and test the revised guide this week.

**Comment by anarcat** [comment]
i did a quite sweep but the changes are large and far-ranging. the "+ more" makes that harder than it could be - for example, there's stuff about IMAP IDLE in there...  this would be easier to review in smaller chunks!

File: pages/security/message-security/openpgp/gpg-best-practices/en.text [hunk 1]
```diff
@@ -61,15 +93,19 @@ bc. keyserver-options no-honor-keyserver-url
...
 
-You should *not* use @gpg --refresh-keys@ or the refresh keys menu item on your email client because you disclose to anyone listening, and the keyserver operator, the whole set of keys that you are interested in refreshing.
+You should *not* use @gpg --refresh-keys@ or the refresh keys menu item on your email client because you disclose to anyone listening, and the keyserver operator, the whole set of keys that you are interested in refreshing. From the [[gnupg documentation -> https://www.gnupg.org/documentation/manuals/gnupg/GPG-Configuration-Options.html#GPG-Configuration-Options]] for @--auto-key-retrieve@:
+
+bq. Note that this option makes a "web bug" like behavior possible. Keyserver or Web Key Directory operators can see which keys you request, so by sending you a message signed by a brand new key (which you naturally will not have on your local keyring), the operator can tell both your IP address and the time when you verified the signature.
+
+Therefor it is a good idea, to [[torify GnuPG requests -> https://trac.torproject.org/projects/tor/wiki/doc/TorifyHOWTO/GnuPG]]: Check that @use-tor@ is present in @~/.gnupg/dirmngr.conf@ (it is default since GnuPG v2.1).
```
Comment (micah) [inline pages/security/message-security/openpgp/gpg-best-practices/en.text]: Therefor should be "Therefore"

File: pages/security/message-security/openpgp/gpg-best-practices/en.text [hunk 2]
```diff
@@ -354,6 +354,14 @@ gpg> save
 
 h2. Additional suggestions.
 
+h3. Consider using an ECC subkey
+
+Since gnupg 2.1 ECC subkeys are supported.
+
+They are faster, which is very useful for security tokens. Note that if there's a RSA and ECC key on a key, GnuPG 1.4 will do the right thing and choose the one that works (e.g. use RSA if it doesn't support ECC).
```
Comment (micah) [inline pages/security/message-security/openpgp/gpg-best-practices/en.text]: "key on a key" is a bit hard to parse. Maybe this could be re-worded to "Note, if there both a RSA and ECC key available, GnuPG ...."

Comment (micah) [reply pages/security/message-security/openpgp/gpg-best-practices/en.text]: Also, you say GnuPG 1.4 does the right thing, this isn't supposed to be 2.1 (i'm unsure, I just see the next section pointing out more details for 2.1)

**Comment by traumschule** [comment]
fixed and rebased. sorry, got held off yesterday.

Didn't test it with amber though because since the latest upgrade of
amber (i guess) something is odd:

```
 amber
server Invalid gemspec in
[/usr/share/rubygems-integration/all/specifications/rbnacl-libsodium.gemspec]:
stack level too deep Invalid gemspec in
[/usr/share/rubygems-integration/2.5.0/specifications/ffi-1.9.10.gemspec]:
stack level too deep Invalid gemspec in
[/usr/share/rubygems-integration/2.5.0/specifications/ffi-1.9.10.gemspec]:
stack level too deep [2018-09-20 08:04:32] INFO  WEBrick 1.4.2
[2018-09-20 08:04:32] INFO  ruby 2.5.1 (2018-03-29) [i386-linux-gnu]
[2018-09-20 08:04:32] INFO  WEBrick::HTTPServer#start: pid=22572
port=8000 [2018-09-20 08:05:25] INFO  Redirect / ==>
http://localhost:8000/en/ 127.0.0.1 - - [20/Sep/2018:08:05:25 CEST]
"GET / HTTP/1.1" 307
0                                                                                                                                       
-
-> / [2018-09-20 08:05:56] INFO  Serving
Page /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77: warning:
Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#pr
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#pr
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#pr ecompiled_method_return_value [2018-09-20 08:08:03]
INFO  Serving
Page /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77: warning:
Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$ ecompiled_method_return_value [2018-09-20 08:17:02]
INFO  Serving
Page /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77: warning:
Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$ ecompiled_method_return_value ^C[2018-09-20 08:24:51]
INFO  going to shutdown ... ^C^C^C^C127.0.0.1 - - [20/Sep/2018:08:16:52
CEST] "GET /en/ HTTP/1.0" 200
0                                                                                                                            
-
-> /en/ 127.0.0.1 - - [20/Sep/2018:08:05:26 CEST] "GET /en/ HTTP/1.1"
200
0                                                                                                                                    
-
-> /en/ 127.0.0.1 - - [20/Sep/2018:08:07:55 CEST] "GET /en/ HTTP/1.1"
200
0                                                                                                                                    
-
-> /en/
Terminated
```

**Comment by kradan** [comment]
fixed and rebased. sorry, got held off yesterday.

Didn't test it with amber though because since the latest upgrade of
amber (i guess) something is odd:

```
 amber
server Invalid gemspec in
[/usr/share/rubygems-integration/all/specifications/rbnacl-libsodium.gemspec]:
stack level too deep Invalid gemspec in
[/usr/share/rubygems-integration/2.5.0/specifications/ffi-1.9.10.gemspec]:
stack level too deep Invalid gemspec in
[/usr/share/rubygems-integration/2.5.0/specifications/ffi-1.9.10.gemspec]:
stack level too deep [2018-09-20 08:04:32] INFO  WEBrick 1.4.2
[2018-09-20 08:04:32] INFO  ruby 2.5.1 (2018-03-29) [i386-linux-gnu]
[2018-09-20 08:04:32] INFO  WEBrick::HTTPServer#start: pid=22572
port=8000 [2018-09-20 08:05:25] INFO  Redirect / ==>
http://localhost:8000/en/ 127.0.0.1 - - [20/Sep/2018:08:05:25 CEST]
"GET / HTTP/1.1" 307
0                                                                                                                                       
-
-> / [2018-09-20 08:05:56] INFO  Serving  
Page /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77: warning:
Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#pr
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#pr
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#pr ecompiled_method_return_value [2018-09-20 08:08:03]
INFO  Serving
Page /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77: warning:
Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$ ecompiled_method_return_value [2018-09-20 08:17:02]
INFO  Serving
Page /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77: warning:
Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$
ecompiled_method_return_value /var/lib/gems/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:77:
warning: Haml::Engine#precompiled_method_return_value
at /usr/lib/ruby/2.5.0/forwardable.rb:157 forwarding to private method
Haml::Compiler#p$ ecompiled_method_return_value ^C[2018-09-20 08:24:51]
INFO  going to shutdown ... ^C^C^C^C127.0.0.1 - - [20/Sep/2018:08:16:52
CEST] "GET /en/ HTTP/1.0" 200
0                                                                                                                            
-
-> /en/ 127.0.0.1 - - [20/Sep/2018:08:05:26 CEST] "GET /en/ HTTP/1.1"  
200
0                                                                                                                                    
-
-> /en/ 127.0.0.1 - - [20/Sep/2018:08:07:55 CEST] "GET /en/ HTTP/1.1"  
200
0                                                                                                                                    
-
-> /en/  
Terminated
```

**Comment by bearbin** [comment]
It would really be good if this received some more attention, the guide on master is now getting rather out of date.

**Comment by kradan** [comment]
@micah @anarcat  thanks for reviewing and sorry for the break! resolved above issues and rebased again. when this is merged, i'll have a deeper look at #451

**Comment by anarcat** [comment]
@kradan there is still a merge conflict here unfortuantely

**Comment by kradan** [comment]
and fixed.

File: pages/security/message-security/openpgp/gpg-best-practices/en.text [hunk 3]
```diff
@@ -18,30 +18,72 @@ Information security is too important to leave to proprietary software. You shou
...
+gpg:                 aka "Kristian Fiskerstrand <kf@gnupg.net>" [unknown]
+gpg:                 aka "Kristian Fiskerstrand <k_f@gentoo.org>" [unknown]
+gpg:                 aka "Kristian Fiskerstrand <kf@sumptuouscapital.com>" [unknown]
+gpg: WARNING: This key is not certified with a trusted signature!
+gpg:          There is no indication that the signature belongs to the owner.
+Primary key fingerprint: 94CB AFDD 3034 5109 5618  35AA 0B7F 8B60 E3ED FAE3
+     Subkey fingerprint: B4EA D120 C7F8 9A4A EA47  2707 250B 7AFE D637 9D85
 
-Then, to use this keyserver pool, you will need to [[download the sks-keyservers.net CA -> https://sks-keyservers.net/sks-keyservers.netCA.pem]], and save it somewhere on your machine. Please remember the path that you save the file to! Next, you should [[verify the certificate's finger print -> https://sks-keyservers.net/verify_tls.php]].
+Congratiolations, you confirmed that you are in possession of an uncorrupted copy of the correct certificate file for the keyserver pool. If you liked that procedure, have a look at [[Riseup's certificates page->certificates]].
```
Comment (ennui93) [inline pages/security/message-security/openpgp/gpg-best-practices/en.text]:
```suggestion
Congratulations, you confirmed that you are in possession of an uncorrupted copy of the correct certificate file for the keyserver pool. If you liked that procedure, have a look at [[Riseup's certificates page->certificates]].
```

**Comment by rolandog** [comment]
Hello. After reading the updated document, I found two possible basic improvements:

I think the URL from `--auto-key-retrieve` in the following line:

https://github.com/riseupnet/riseup_help/blob/e28abf649a6996047c59974893e78636f40099f0/pages/security/message-security/openpgp/gpg-best-practices/en.text#L62

Can point towards the corresponding entry:

```
[[@--auto-key-retrieve@ -> https://www.gnupg.org/documentation/manuals/gnupg/GPG-Configuration-Options.html#index-auto_002dkey_002dretrieve]]
```

Also, should `[@--sign-key@]` in this line be an external link as well?

https://github.com/riseupnet/riseup_help/blob/e28abf649a6996047c59974893e78636f40099f0/pages/security/message-security/openpgp/gpg-best-practices/en.text#L149

If so, it may point towards:

```
[[@--sign-key@ -> https://www.gnupg.org/documentation/manuals/gnupg/OpenPGP-Key-Management.html#index-sign_002dkey]]
```

File: pages/email/clients/k9/de.text [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 p(pull-right). !(image-right)K9mail.png!
 
-*K-9 Mail* ist eine Freie Software E-Mail-Anwendung für Android. Die Anwendung ist ein funktionalerer Ersatz für das vorinstallierte E-Mailprogramm der meisten Telefone. Es unterstützt POP3- und IMAP-Postfächer. Es kann von "Google Play":https://f-droid.org und der "K-9-Webseite":https://code.google.com/p/k9mail heruntergeladen werden.
+[[*K-9 Mail* -> https://k9mail.github.io]] ist ein freies Emailprogramm für Android. Die Anwendung ist ein funktionalerer Ersatz für das vorinstallierte Emailprogramm der meisten Telefone. Es unterstützt POP3- und IMAP-Postfächer, sowie [[IMAP IDLE->email/clients#imap-idle]].
```
Comment (jonassmedegaard) [inline pages/email/clients/k9/de.text:5]: K-9 Mail now has a proper non-Github domain: https://k9mail.app/

**Comment by disturbio** [comment]
Hi,

It took a long time to get into this commit, but one of the reasons is that GnuPG has been incorporating sane config options by default and this document was less and less needed. After the keyserver doom and reborn this guide became harmful. Now there is a deprecation warning for the document and we will close this PR.

I'm very sorry for the slow response and bad management for this request, but as time is limited and too much things to take care we let it sleep for too long.

I'm closing this for now and we will review and things what things we should incorporate from the other edited pages
